ovh/helpers/helpers.go

```diff
@@ -358,3 +358,25 @@ func ValidateSubsidiary(v string) error {
 		"ws",
 	})
 }
+
+// HashMapRegionAttributes creates an hash for the region attributes.
+func HashMapRegionAttributes(v interface{}) int {
+	attributes, ok := v.(map[string]interface{})
+	if !ok {
+		return 0
```

Comment:
Why are you returning 0?
In which case should we obtain ok == false?
Is it possible to return an error message or something?
Thanks

Reply:
Actually, I'm returing 0 to be conformed to the [hashing function signature](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema#SchemaSetFunc), that is `func(interface{}) [int](https://pkg.go.dev/builtin#int)`. Said that, the conversion check regarding the `v interface` is just to ensure that everything goes well but, in case of any failure, the only returning option for this signature is an integer and not an error. Maybe, a `-1` is a better option in place of returning `0`. 

However, since the schema should not change that conversion check is just a stretch. Let me know the way is better to modify. Also, have a code like this is an option:

```golang
// HashMapRegionAttributes creates an hash for the region attributes.
func HashMapRegionAttributes(v interface{}) int {
	attributes := v.(map[string]interface{})
	builder := strings.Builder{}

	for _, key := range []string{"status", "region", "openstackid"} {
		value, inMap := attributes[key]
		if inMap {
			stringValue, ok := value.(string)
			if ok {
				builder.WriteString(stringValue)
			}
		}
	}

	return schema.HashString(builder.String())
}
```

Reply:
Currently, I opted for that last option. I hope it is the best way to approach the problem.